### ISSUE TITLE
Add a RecordModel.getMapValue() method

### ISSUE THREAD

**Issue by PeetoomHeida** [pocketbase/dart-sdk#45]
I have run into a headache while trying to work with a database. Some of the fields store json data, which I would like to access directly. The methods getStringValue() etc are super useful for pulling other fields, but getting the value out of a field that holds json is much more involved. It would be great if there was a method for the RecordModel class to directly access these fields, returning a Map<String, dynamic>. Perhaps this isn't technically feasible, but I thought I would suggest it just in case.

**Comment by ganigeorgiev** [comment]
I'll consider adding a new generic helper `getDataValue<T>(key, [defaultValue])` with the next release (_the existing accessors could be defined as aliases of it, eg. `getStringValue(field)` will be the same as `getDataValue<String>(field)`_).

For convenience we can also allow the key to be a dot-notation path to allow more easier access to nested object values. An example pseudo-code:

```dart
final data = final data = {"a": {"b": [{"b1": 1}, {"b2": 2}, {"b3": 3}]}};
final record = RecordModel(data: data);

record.getDataValue<int>("a.b.c", "missing"); // "missing"
record.getDataValue<int>("a.b.2.b3"); // 3
record.getDataValue<String>("a.b.2.b3"); // "3"
```

**Comment by ganigeorgiev** [comment]
The above is available with [v0.14.0 release](https://pub.dev/packages/pocketbase/changelog).

**Comment by PeetoomHeida** [comment]
You are awesome! Thanks so much!
